source/agora/common/TransactionPool.d

```diff
@@ -80,6 +82,20 @@ public class TransactionPool
         // create the table if it doesn't exist yet
         this.db.execute("CREATE TABLE IF NOT EXISTS tx_pool " ~
             "(key BLOB PRIMARY KEY, val BLOB NOT NULL)");
+
+        // insert Input objects to input_map from the transactions stored in DB.
```

Comment:
We could serialize the map (or Set) to disk and load it on start-up instead of recreating it each time the transaction pool is loaded. Iterating over the entire transaction pool on load can be costly.

Reply:
I think there are some issues about serializing the set to disk (I'am now changing the data structure using Set!Hash as you suggested). First, There is a possibility that the set of Input hashes and the database do not have same information in case a node crashes before serializing the hash set. Second, The tx pool database already have the data for hash set for Input objects, so it's kind of redundant. I know your concerns about loading the Input hashes by iterating over the all transactions, but I don't think it's so big. 
Any opinion would be helpful.

Reply:
Ok, then for starters keep it the way it is. We can optimize it later, and ensure we add checksums and things like that later on.